### PR TITLE
Adding unit tests only for debug build type

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### Description

This PR changes our github actions so we only run unit tests for the debug build type.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes. **Not necessary**
- [ ] New code is covered by unit tests. **Not necessary**
- [ ] Comparison screenshots added for visual changes. **Not necessary**
- [x] Reviewers added

### CODE REVIEW

**Comment by zsmb13** [comment]
What's the benefit of making this change?

**Comment by leandroBorgesFerreira** [comment]
@zsmb13  This CI would run faster. There's no need to unit test both debug and release build types since there's difference between then in our code

**Comment by zsmb13** [comment]
I mean, there _is_ a difference, and we've had failing release builds before due to never building that variant. But for tests, it's probably enough to run debug, agreed.

Though I don't see any speedup in the CI run on this PR compared to other ones, still takes 7 minutes overall. Might be worth looking into.

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
       - name: ktlint
         run: ./gradlew :stream-chat-android:ktlintCheck :stream-chat-android-sample:ktlintCheck
       - name: Unit tests
-        run: ./gradlew :stream-chat-android:test :stream-chat-android-sample:test --stacktrace
+        run: ./gradlew :stream-chat-android:testDebug :stream-chat-android-sample:testDebug --stacktrace
```
Comment (zsmb13) [inline .github/workflows/pr-checks.yml:81]: We should remove the uploading of the release test results if we make this change, it's a couple lines down from here in a step mistakenly named `Upload testDebugUnitTest results` due to a copy-paste error 😄

Comment (leandroBorgesFerreira) [reply .github/workflows/pr-checks.yml:75]: done

Comment (zsmb13) [reply .github/workflows/pr-checks.yml:75]: This is there for all different jobs, not just the one, we should remove it from all of them.

Comment (leandroBorgesFerreira) [reply .github/workflows/pr-checks.yml:69]: Oh, I see. Removed from all jobs now.

**Review by zsmb13 — DISMISSED** [review]

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**